cataclysmbnteam/Cataclysm-BN
# Temperature bugfixes: eternal season and clothing stacking

#### Summary



SUMMARY: Bugfixes "Keep season temperature constant when eternal seasons are on"

#### Purpose of change



Fixes #1373

Eternal seasons only affect things that directly depend on calculated season, but weather depends only on year fraction, not season. Here I make year fraction constant when eternal seasons are on, making weather depend only on hour of the day and random noise.

Fixes #1417

Clothing warmth stacking was broken, it used `max` instead of sum.

#### Describe the solution



For purposes of weather, when eternal seasons are on, the year fraction is set to middle of the season.

Also some refactoring to use temperature units in more places where Fahrenheit was used.

Minor bonus: can change eternal season in debug menu.

#### Describe alternatives you've considered



#### Testing



1. Check that 2 scarves are warmer than 1 scarf
2. Check, by changing the time a bunch of times, that when eternal winter is on, temperatures on thermometer don't go above -1C (-14C base, +5C max daily offset +8C max noise offset).
3. Check what when eternal summer is on, temperatures on thermometer don't go below +3C (16C base - 13C offsets).

Tests 2 and 3 are automatic, but don't actually pass for some reason. I added a 3C margin and now it passes. I should fix it, but for now, the changes fix more serious problems, so math that mysteriously goes up or down by 3C is still an improvement.

#### Additional context
